File: lambdaconscarcdr.php
#!/usr/bin/env php

<?php
$kons  = function ($hd, $tl) {
    return function ($x) use ($hd, $tl) {
        return $x ? $hd : $tl;
    };};

$virst = function ($l) { return $l(true ); };
$rrest = function ($l) { return $l(false); };

$lstt  = $kons(1, $kons(2, null));

$virst($rrest($lstt));
?>

File: lambdaconscarcdr.rb
#!/usr/bin/env ruby

Tru   = ->(thn, _  ) { thn }
Fls   = ->(_  , els) { els }

Kons  = ->(hd, tl) { -> x { x.(hd, tl) }}
Virst = -> l { l.(Tru) }
Rrest = -> l { l.(Fls) }

Lstt  = Kons.(1, Kons.(2, nil))

Virst.(Rrest.(Lstt))
